src/components/WaveformErrorBoundary.js

```diff
@@ -0,0 +1,40 @@
+import React, { Component } from 'react';
+import { Alert } from 'react-bootstrap';
+
+class WaveformErrorBoundary extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { hasError: false };
+  }
+
+  state = {
+    showAlert: false
+  };
+
+  static getDerivedStateFromError(error) {
+    // Update state so the next render will show the fallback UI.
+    return { hasError: true, showAlert: true };
+  }
```

Comment:
@adamjarling This is awesome! :+1: I didn't know this existed.
But it doesn't hit this part of the code after the Waveform network request returns an error.

Reply:
Yea, something is off on this.  I noticed I had 2 branches locally for this:)   Will tidy up and re-push today.

Reply:
In general looks like React's Error Boundaries are used for `render()` errors, and not for event handling or async network communication.  However if our network request code is returning some type of error object, maybe that'd trigger it?  I'll play a bit more with this.

Reply:
Ok, so re-reading your comment from above about this PR not handling the network request error, you're correct.  This particular error boundary handles any other errors which might occur  only in the `Waveform.js` component, during user interaction with the editor.  ie:
```
<WaveformErrorBoundary>
          <Waveform
            waveformRef={ref => (this.waveformContainer = ref)}
            mediaPlayerRef={ref => (this.mediaPlayer = ref)}
          />
</WaveformErrorBoundary>
```

The network request itself lives in `WaveformContainer.js`, and seems to be handled ok by the `try {} catch(e) {}` block?

Reply:
Oh okay :+1: Yeah the network request errors go through `handleResponse()` method, which triggers alerts from `AlertDismissable` component.